tokio/src/runtime/local_runtime/runtime.rs

```diff
@@ -0,0 +1,375 @@
+    ///
+    /// # Examples
+    ///
+    /// Creating a new `LocalRuntime` with default configuration values.
+    ///
+    /// ```
+    /// use tokio::runtime::LocalRuntime;
+    ///
+    /// let rt = LocalRuntime::new()
+    ///     .unwrap();
+    ///
+    /// // Use the runtime...
+    /// ```
+    ///
+    /// [mod]: crate::runtime
+    /// [runtime builder]: crate::runtime::Builder
+    pub fn new() -> std::io::Result<LocalRuntime> {
+        Builder::new_current_thread()
+            .enable_all()
+            .build_local(&mut Default::default())
+    }
+
+    /// Returns a handle to the runtime's spawner.
+    ///
+    /// The returned handle can be used to spawn tasks that run on this runtime, and can
+    /// be cloned to allow moving the `Handle` to other threads.
+    ///
+    /// Local tasks cannot be spawned on this handle.
+    ///
+    /// Calling [`Handle::block_on`] on a handle to a `LocalRuntime` is error-prone.
+    /// Refer to the documentation of [`Handle::block_on`] for more.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::runtime::LocalRuntime;
+    ///
+    /// let rt = LocalRuntime::new()
+    ///     .unwrap();
+    ///
+    /// let handle = rt.handle();
+    ///
+    /// // Use the handle...
+    /// ```
+    pub fn handle(&self) -> &Handle {
+        &self.handle
+    }
+
+    /// Spawns a future onto the `LocalRuntime`.
+    ///
+    /// See the documentation for the equivalent method on [Runtime] for more information
+    ///
+    /// [Runtime]: crate::runtime::Runtime::spawn
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::runtime::LocalRuntime;
+    ///
+    /// # fn dox() {
+    /// // Create the runtime
+    /// let rt = LocalRuntime::new().unwrap();
+    ///
+    /// // Spawn a future onto the runtime
+    /// rt.spawn(async {
+    ///     println!("now running on a worker thread");
+    /// });
+    /// # }
+    /// ```
+    #[track_caller]
+    pub fn spawn<F>(&self, future: F) -> JoinHandle<F::Output>
+    where
+        F: Future + Send + 'static,
+        F::Output: Send + 'static,
+    {
+        if cfg!(debug_assertions) && std::mem::size_of::<F>() > BOX_FUTURE_THRESHOLD {
+            self.handle.spawn_named(Box::pin(future), None)
+        } else {
+            self.handle.spawn_named(future, None)
+        }
+    }
+
+    /// Spawns a task which isn't `!Send + Sync` on the runtime.
+    #[track_caller]
+    pub fn spawn_local<F>(&self, future: F) -> JoinHandle<F::Output>
+    where
+        F: Future + 'static,
+        F::Output: 'static,
+    {
+        // safety: spawn_local can only be called from `LocalRuntime`, which this is
+        unsafe {
+            if cfg!(debug_assertions) && std::mem::size_of::<F>() > BOX_FUTURE_THRESHOLD {
+                self.handle.spawn_local_named(Box::pin(future), None)
+            } else {
+                self.handle.spawn_local_named(future, None)
+            }
+        }
+    }
```

Comment:
Hmm. These methods are identical. Perhaps just have a `spawn` with the signature of `spawn_local`?